src/reducers/agentReducer.js

```diff
@@ -8,8 +8,29 @@ export default function agentReducer(state=initialState.agent, action) {
         case types.AGENT_INFO_FETCH_SUCCEED:
             return action.agent;
 
+        //add new quick reply successfully
+        case types.ADD_QUICK_REPLY_SUCCEED:
+            return {...state, replies: [...state.replies, action.reply]}
+
+        case types.DELETE_QUICK_REPLY_SUCCEED:
+            return {...state, replies: state.replies.filter(reply => parseInt(reply.id) !== parseInt(action.replyId))}
+
+        case types.UPDATE_QUICK_REPLY_SUCCEED:
+
+            return {...state, replies: state.replies.map(reply => {
+                console.log(parseInt(reply.id) !== parseInt(action.reply.id));
+                if (parseInt(reply.id) !== parseInt(action.reply.id) ) {
+                    return reply;
+                }
+                return action.reply;
+
+            })};
+
         // default case, return current state
         default:
             return state;
+
+
     }
+
 }
```

Comment:
C config editor cho có newline at the end of file nhé

Reply:
Cái này là t follow standard thôi. t thấy ngôn ngữ nào họ cũng có chuẩn yêu cầu là có new line at the end of file. Nên cứ follow nhé.